src/test/kotlin/TaskTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.dlfsystems
+
+import kotlinx.coroutines.runBlocking
+import kotlin.test.Test
+
+class TaskTest: YeggTest() {
+
+    @Test
+    fun `Suspend`() = runBlocking {
+        runForOutput($$"""
+            foo = 20
+            startTime = $sys.time
+            suspend(2)
+            elapsedTime = $sys.time - startTime
+            notifyConn("$foo in $elapsedTime seconds")
+        """, """
+            20 in 2 seconds
+        """)
+    }
+
+    @Test
+    fun `Fork`() = runBlocking {
+        runForOutput($$"""
+            foo = 20
+            startTime = $sys.time
+            fork (1) {
+                elapsed = $sys.time - startTime
+                foo *= 2
+                notifyConn("$foo in $elapsed seconds")
+            }
+            fork (2) {
+                elapsed = $sys.time - startTime
+                foo *= 3
+                notifyConn("$foo in $elapsed seconds")
+                fork (1) {
+                    elapsed = $sys.time - startTime
+                    foo = "ACK"
+                    notifyConn("$foo in $elapsed seconds")
+                }
+            }
+            notifyConn("start $foo")
+        """, """
+            start 20
+            40 in 1 seconds
+            60 in 2 seconds
+            ACK in 3 seconds
```

Comment:
I guess this might not work if you use `advanceUntilIdle()`.  But maybe?  I'm not sure what counts as 'idle'.